iceoryx_posh/include/iceoryx_posh/internal/roudi/roudi_multi_process.hpp

```diff
@@ -41,17 +41,33 @@ using namespace iox::units::duration_literals;
 class RouDiMultiProcess
 {
   public:
+
+    // indicate whether the message queue thread will start directly or deferred
+    // this is important for derived classes which may need to initialize their members before the thread starts
+    enum class MQThreadMode {
+      START,
+      DEFER_START
+    };
+   
+
     RouDiMultiProcess& operator=(const RouDiMultiProcess& other) = delete;
     RouDiMultiProcess(const RouDiMultiProcess& other) = delete;
 
     RouDiMultiProcess(RouDiMemoryInterface& roudiMemoryInteface,
                       PortManager& portManager,
                       const MonitoringMode f_monitoringMode = MonitoringMode::ON,
-                      const bool f_killProcessesInDestructor = true);
+                      const bool f_killProcessesInDestructor = true,
```

Comment:
Very minor but the variable is inconsistent with the `.cpp`.
`killProcessesInDestructor` vs `f_killProcessesInDestructor`.